simulations/vip-152/vip-152/simulation.ts

```diff
@@ -0,0 +1,136 @@
+      psr = new ethers.Contract(PSR, PSR_ABI, provider);
+    });
+
+    it("validate admin", async () => {
+      expect(await vBNB.admin()).to.be.equal(NORMAL_TIMELOCK);
+    });
+
+    it("config check", async () => {
+      expect(await psr.totalDistributions()).to.be.equal(0);
+    });
+
+    it("ownership check", async () => {
+      expect(await psr.pendingOwner()).to.be.equal(NORMAL_TIMELOCK);
+    });
+  });
+
+  testVip("VIP-152", vip152(), {
+    callbackAfterExecution: async txResponse => {
+      await expectEvents(txResponse, [PSR_ABI], ["DistributionConfigAdded"], [4]);
+      await expectEvents(txResponse, [PSR_ABI], ["PoolRegistryUpdated"], [1]);
+    },
+  });
+
+  describe("Post-VIP behavior", async () => {
+    let vBNB: ethers.Contract;
+    let psr: ethers.Contract;
+    let vBNBAdmin: ethers.Contract;
+    let WBNB: ethers.Contract;
+    let proxyAdmin: ethers.Contract;
+
+    before(async () => {
+      impersonateAccount(NORMAL_TIMELOCK);
+      const signer = await ethers.getSigner(NORMAL_TIMELOCK);
+
+      vBNB = new ethers.Contract(vBNB_ADDRESS, vBNB_ABI, provider);
+      psr = new ethers.Contract(PSR, PSR_ABI, signer);
+      vBNBAdmin = new ethers.Contract(VBNBAdmin_ADDRESS, vBNBAdmin_ABI, signer);
+      WBNB = new ethers.Contract(WBNB_ADDRESS, ERC20_ABI, provider);
+      proxyAdmin = new ethers.Contract(PROXY_ADMIN, PROXY_ADMIN_ABI, provider);
+    });
+
+    it("validate admin", async () => {
+      expect(await vBNB.admin()).to.be.equal(VBNBAdmin_ADDRESS);
+    });
+
+    it("validate PSR", async () => {
+      expect(await vBNBAdmin.protocolShareReserve()).to.be.equal(PSR);
+    });
+
+    it("config check", async () => {
+      expect(await psr.totalDistributions()).to.be.equal(4);
+    });
+
+    it("ownership check", async () => {
+      expect(await psr.pendingOwner()).to.be.equal("0x0000000000000000000000000000000000000000");
+      expect(await psr.owner()).to.be.equal(NORMAL_TIMELOCK);
+    });
+
+    it("configurations", async () => {
+      let config = await psr.distributionTargets(0);
+      expect(config.schema).to.be.equal(0);
+      expect(config.percentage).to.be.equal(50);
+      expect(config.destination).to.be.equal(RISK_FUND);
+
+      config = await psr.distributionTargets(1);
+      expect(config.schema).to.be.equal(0);
+      expect(config.percentage).to.be.equal(50);
+      expect(config.destination).to.be.equal(TREASURY);
+
+      config = await psr.distributionTargets(2);
+      expect(config.schema).to.be.equal(1);
+      expect(config.percentage).to.be.equal(40);
+      expect(config.destination).to.be.equal(RISK_FUND);
+
+      config = await psr.distributionTargets(3);
+      expect(config.schema).to.be.equal(1);
+      expect(config.percentage).to.be.equal(60);
+      expect(config.destination).to.be.equal(TREASURY);
+    });
+
+    it("reduce reserves", async () => {
+      expect(await WBNB.balanceOf(PSR)).to.be.equal(ethers.utils.parseEther("0"));
```

Comment:
I think that at this point we are interacting with the contracts with the Normal Timelock contract (we are impersonating it in the `before` function). I would impersonate a regular account instead at the beginning of this test (and I would re-impersonate the Normal Timelock to avoid unexpected side effects on other tests).